### PR TITLE
Export egoposes as latlon in json or kml format

### PR DESCRIPTION
**Pull request by holger-nutonomy**
This code was originally contributed by @jean-lucas https://github.com/nutonomy/nuscenes-devkit/pull/347.
The modified version allows us to export either json or kml files. Kml files can be conveniently visualized in most mapping programs (incl. Google Earth). The trajectories were visually verified to be "correct" (on the road).
singapore

The kml files for trainval and mini can be found at: [latlon_v1.0-.zip](https://github.com/nutonomy/nuscenes-devkit/files/4442879/latlon_v1.0-.zip)

### CODE REVIEW

File: python-sdk/nuscenes/scripts/export_poses.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+import os
+from typing import List, Tuple, Dict
+
+from tqdm import tqdm
+
+from nuscenes.nuscenes import NuScenes
+
+
+EARTH_RADIUS_METERS = 6.378137e6
+REFERENCE_COORDINATES = {
```
Comment (venice-erin) [inline python-sdk/nuscenes/scripts/export_poses.py:21]: how are these obtained?

File: python-sdk/nuscenes/scripts/export_poses.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    return pose_list
+
+
+def get_coordinate(ref_lat: float, ref_lon: float, bearing: float, dist: float) -> Tuple[float, float]:
+    """
+    Using a reference coordinate, extract the coordinates of another point in space given its distance and bearing
+    to the reference coordinate. For reference, please see: https://www.movable-type.co.uk/scripts/latlong.html.
+    :param ref_lat: Latitude of the reference coordinate in degrees, ie: 42.3368.
+    :param ref_lon: Longitude of the reference coordinate in degrees, ie: 71.0578.
+    :param bearing: The angle the target point makes with the reference coordinate in radians, clockwise from north.
```
Comment (venice-erin) [inline python-sdk/nuscenes/scripts/export_poses.py:58]: `The angle that the`?

File: python-sdk/nuscenes/scripts/export_poses.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    :param dist: The distance (in meters from the reference point to the target point.
```
Comment (venice-erin) [inline python-sdk/nuscenes/scripts/export_poses.py:59]: missing closing parenthesis.

File: python-sdk/nuscenes/scripts/export_poses.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    """
+    For each pose value, extract its respective lat/lon coordinate and timestamp.
+    
+    This makes the following two assumptions in order to work:
+        1. The reference coordinate for each map is in the south-western corner.
+        2. The origin of the global poses is also in the south-western corner (and identical to 1).
+
+    :param location: The name of the map the poses correspond to, ie: 'boston-seaport'.
+    :param poses: All pose dictionaries of a scene.
+    :return: A list of lat/lon coordinate dicts for each pose.
```
Comment (venice-erin) [inline python-sdk/nuscenes/scripts/export_poses.py:86]: did not mention about timestamp in return.

File: python-sdk/nuscenes/scripts/export_poses.py [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+def derive_latlon(location: str, poses: List[dict]) -> List[Dict[str, float]]:
+    """
+    For each pose value, extract its respective lat/lon coordinate and timestamp.
+    
+    This makes the following two assumptions in order to work:
+        1. The reference coordinate for each map is in the south-western corner.
+        2. The origin of the global poses is also in the south-western corner (and identical to 1).
+
+    :param location: The name of the map the poses correspond to, ie: 'boston-seaport'.
+    :param poses: All pose dictionaries of a scene.
```
Comment (venice-erin) [inline python-sdk/nuscenes/scripts/export_poses.py:85]: perhaps define what is the dict here..`Dict[str, Any]`?  What are the strings here?

File: python-sdk/nuscenes/scripts/export_poses.py (hunk 1 above)
Comment (holger-nutonomy) [reply python-sdk/nuscenes/scripts/export_poses.py:21]: These can be computed from the slippy map tile system, which tiles the globe into a grid at a specified resolution: https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames

File: python-sdk/nuscenes/scripts/export_poses.py (hunk 5 above)
Comment (holger-nutonomy) [reply python-sdk/nuscenes/scripts/export_poses.py]: It's a standard egopose dict, so I just mention that.

**Pull request merged**